### PR TITLE
bnc 901598

### PR DESCRIPTION
**Pull request by wfeldt**
This improved driver update handling a bit.

### CODE REVIEW

File: url.c [hunk 1]
```diff
@@ -1463,6 +1463,122 @@ int url_mount(url_t *url, char *dir, int (*test_func)(url_t *))
...
+
+  if(util_check_exist(file) != 'r') {
+    if(config.debug) fprintf(stderr, "%s: gpg check = %d\n", file, err);
+
+    return err;
+  }
+
+  strprintf(&cmd,
+    "gpg --homedir /root/.gnupg --batch --no-default-keyring --keyring /installkey.gpg "
+    "--ignore-valid-from --ignore-time-conflict --output %s.unpacked %s 2>&1",
```
Comment (aschnell) [inline url.c]: If file can contain spaces or other bad chars shell escaping is needed.

File: url.c [hunk 2]
```diff
@@ -1463,6 +1463,122 @@ int url_mount(url_t *url, char *dir, int (*test_func)(url_t *))
...
+  );
+
+  if((f = popen(cmd, "r"))) {
+    while(getline(&buf, &len, f) > 0) {
+      if(config.debug >= 2) fprintf(stderr, "%s", buf);
+      if(strncmp(buf, "gpg: Signature made", sizeof "gpg: Signature made" - 1)) is_sig = 1;
+      if(strncmp(buf, "gpg: Good signature", sizeof "gpg: Good signature" - 1)) sig_ok = 1;
+    }
+    err = pclose(f) ? 1 : 0;
+    if(config.debug >= 2) fprintf(stderr, "gog returned %s\n", err ? "an error" : "ok");
```
Comment (aschnell) [inline url.c]: gog -> gpg

File: url.c (hunk 1 above)
Comment (wfeldt) [inline url.c]: file is a temporary file with a generated name; I've how put quotes around it anyway

**Comment by aschnell** [comment]
AFAIK the changelog is generated from the git history.

LGTM - although not easy to read.

**Pull request merged**